handlers_global.go

```diff
@@ -75,31 +80,41 @@ func handleImport(s *Server) http.Handler {
 		// that is not empty (ie, all fields are the zero value)
 		// because that is usually the sign that we are unmarshalling
 		// into the wrong struct type
-		var nonEmpty bool
-		sentinel := samplers.JSONMetric{}
-		for _, metric := range jsonMetrics {
-			if !reflect.DeepEqual(sentinel, metric) {
-				// we have found at least one entry that is properly formed
-				nonEmpty = true
-				break
-			}
-		}
 
-		if !nonEmpty {
+		if !s.nonEmpty(trace.Attach(ctx), jsonMetrics) {
 			const msg = "Received empty or improperly-formed metrics"
 			http.Error(w, msg, http.StatusBadRequest)
+			trace.Status = ssf.SSFSample_CRITICAL
 			innerLogger.Error(msg)
 			return
 		}
 
 		w.WriteHeader(http.StatusAccepted)
 		s.statsd.TimeInMilliseconds("import.response_duration_ns",
-			float64(time.Now().Sub(start).Nanoseconds()),
+			float64(time.Now().Sub(trace.Start).Nanoseconds()),
 			[]string{"part:request", fmt.Sprintf("encoding:%s", encoding)},
 			1.0)
 
 		// the server usually waits for this to return before finalizing the
 		// response, so this part must be done asynchronously
-		go s.ImportMetrics(jsonMetrics)
+		go s.ImportMetrics(trace.Attach(ctx), jsonMetrics)
 	})
 }
+
+// nonEmpty returns true if there is at least one non-empty
+// metric
+func (s *Server) nonEmpty(ctx context.Context, jsonMetrics []samplers.JSONMetric) bool {
+
+	trace := trace.SpanFromContext(ctx)
+	defer trace.Record("veneur.import.nonEmpty.trace", nil)
```

Comment:
TRACE ALL THE THINGS

also, should we be marking this trace as critical if it fails (as opposed to the parent)?